pydantic/functional_serializers.py

```diff
@@ -217,6 +259,30 @@ def model_serializer(
 ) -> Callable[[Any], Any]:
     """Decorator that enables custom model serialization.
 
+    This is useful when a model need to be serialized in a customized manner, allowing for flexibility beyond just specific fields.
+
+    Following [`field_serializer`][pydantic.functional_serializers.field_serializer] example, we not only want to sort courses field but also capitalize the name of student.
+
+    ```python
+    from typing import Set
+
+    from pydantic import BaseModel, model_serializer
+
+    class StudentModel(BaseModel):
+        name: str
+        courses: Set[str]
+
+        @model_serializer(when_used='json')
+        def serialize_model(self):
+            return {'name': self.name.title(), 'courses': sorted(self.courses)}
+
+    student = StudentModel(
+        name='jane doe', courses={'Math', 'Chemistry', 'English'}
+    )
+    print(student.model_dump_json())
+    #> {"name":"Jane Doe","courses":["Chemistry","English","Math"]}
+    ```
+
```

Comment:
I like the thought here, but I think we probably want to use an example that couldn't be achieved by using 2 different `field_serializer` decorators.

For example, these examples from the usage docs are more `model_serializer` specific (there's one relevant example per section linked below)
* https://docs.pydantic.dev/latest/concepts/serialization/#overriding-the-return-type-when-dumping-a-model
* https://docs.pydantic.dev/latest/concepts/serialization/#custom-serializers

Reply:
I've rewrited it with another example.

https://github.com/pydantic/pydantic/blob/fb0b40a45cc1d75318e4a809eb72c5460e8fe5cd/pydantic/functional_serializers.py#L264-L284